Thread-device.md

```diff
@@ -0,0 +1,92 @@
+## How to Set Up a Matter Thread Lighting Device on Nordic nRF52840 Dongle
+
+This guide demonstrates how to build and flash a Matter Thread lighting application firmware onto a Nordic Semiconductor nRF52840 Dongle, 
+thereby transforming the dongle into a Matter Thread device.
+
+Complete the following steps to build and flash the firmware:
+
+### Environment Setup
+- Ubuntu 23.04 OS
+- nRF52840 dongle
+
+### Version tested in this guide
+- Nordic Semiconductor's nRF Connect for Desktop: 4.2.0
+- Nordic Semiconductor's nRF Connect SDK: 2.4.2
+- [Matter](https://github.com/project-chip/connectedhomeip) SDK: 1.1.0.1
+
+### 1. Setting up the Build Environment for nRF Connect SDK and Matter SDK
+
+First, follow [Using native shell for setup](https://github.com/project-chip/connectedhomeip/tree/v1.1.0.1/examples/lighting-app/nrfconnect#using-native-shell-for-setup)
+guide's steps 1 and 2 to install `nRF command-line tools`, `GN meta-build system`, and `nRF Connect SDK`.
+
+Once nRF Desktop is installed, change its permissions, install dependencies, and run it:
+```bash
+chmod +x nrfconnect-4.2.0-x86_64.appimage
+sudo apt install libfuse2
+./nrfconnect-4.2.0-x86_64.appimage
+```
+
+In nRF Connect for Desktop, install the latest `toolchain manager` and `nRF Connect SDK v2.4.2`. 
+After installations, click the down arrow next to the nRF Connect SDK version you installed and select `Open Terminal` or `Open Bash`.
+
+### 2. Installing and Activating Matter SDK:
+Install the Matter SDK with a shallow clone for the Linux platform:
+```bash
+cd ~
+git clone https://github.com/project-chip/connectedhomeip.git --depth=1 --branch=v1.1.0.1
+cd ~/connectedhomeip/
+scripts/checkout_submodules.py --shallow --platform linux
+git submodule update --init
```

Comment:
These two commands work against each other. The first command makes a shallow clone of Linux platform projects, the second makes a deep clone of everything.

I'll do some further testing to see if the first command has any extra effect.

Reply:
Does `git submodule update --init` perform a deep clone?  It only initializes and updates submodules. 

 a deep clone of submodules will be `git submodule update --init --recursive`

Reply:
By setting `--recursive`, it will update also the nested submodules; it isn't about the history depth. 
The shallow/deep history isn't affecting functionality for the build, but just the size of the history.

Anyway, the second command clones all submodules and the first one just the ones for Linux platform. What we want here is indeed all modules, or only those relevant to this build.